Prog/Demographic trends.sas

```diff
@@ -0,0 +1,1458 @@
+"2401004",
+"2401005",
+"2401006",
+"2401007",
+"2401101",
+"2401102",
+"2401103",
+"2401104",
+"2401105",
+"2401106",
+"2401107",
+"5100101",
+"5100100",
+"5100301",
+"5100302",
+"5100303",
+"5100304",
+"5100305",
+"5100600",
+"5100501",
+"5100502",
+"5100200"
+)));;
+set Ipums.Ipums_2000_dc(where=(gq in (1,2))) Ipums.Ipums_2000_va(where=(gq in (1,2))) Ipums.Ipums_2000_md(where=(gq in (1,2)));
+keep pernum gq upuma hhwt perwt year serial numprec related notnonrelate numberofpop spouse;
+
+if related in (1113, 1115, 1241, 1260, 1270, 1301) then notnonrelate=0;
+else if related in (201, 301, 302, 303, 401, 501, 601, 701, 801, 901, 1001, 1011, 1021, 1031, 1041, 1242) then notnonrelate=1;
+
+if related in ( 201) then spouse=1;
+
+numberofpop=1;
+
+run;
+
+proc summary DATA=hhrelate_2000;
+  class serial;
+  var notnonrelate numberofpop spouse;
+  output out =aaa_2000 sum= ;
+run;
+
+data nonrelatehh_2000 ;
+set aaa_2000;
+if numberofpop>=2 then do;
+	if notnonrelate>=1 then nonrelatehh=0;
+	else if notnonrelate=0 then nonrelatehh=1;
+	else nonrelatehh=.;
+end;
+if numberofpop=1 then single=1;
+else single=0;
+
+if numberofpop=2 & spouse=1 then couplefam=1;
+else couplefam=0;
+
+run;
+
+%macro hhtype_1(year);
+data hhtype_1_&year. (where=(upuma in ("1100101", "1100102", "1100103", "1100104", "1100105", "2401600", "2400301", "2400302","2401001", "2401002","2401003", "2401004", "2401005", "2401006", "2401007", "2401101", "2401102", "2401103", "2401104", "2401105", "2401106", "2401107","5101301", "5101302", "5159301", "5159302", "5159303", "5159304", "5159305","5159306", "5159307", "5159308", "5159309", "5110701", "5110702" , "5110703", "5151244","5151245", "5151246", "5151255"))) ;
+set Ipums.ACS_&year._dc(where=(pernum=1 and gq in (1,2))) Ipums.ACS_&year._va(where=(pernum=1 and gq in (1,2))) Ipums.ACS_&year._md(where=(pernum=1 and gq in (1,2)));
+keep pernum gq upuma Jurisdiction hhwt perwt year serial numprec HHINCOME HHTYPE relate incomecat;
+
+
+	%assign_jurisdiction; 
+
+	  if hhincome ~=.n or hhincome ~=9999999 then do; 
+		 %dollar_convert( hhincome, hhincome_a, &year., 2017, series=CUUR0000SA0 )
+	   end; 
+  
+		if hhincome_a in ( 9999999, .n , . ) then incomecat=.;
+			else do; 
+			    if hhincome_a<=32600 then incomecat=1;
+				else if 32600<hhincome_a<=54300 then incomecat=2;
+				else if 54300<hhincome_a<=70150 then incomecat=3;
+				else if 70150<hhincome_a<=108600 then incomecat=4;
+				else if 108600<hhincome_a<=130320 then incomecat=5;
+				else if 130320<hhincome_a<=217200 then incomecat=6;
+				else if 217200 < hhincome_a then incomecat=7;
+			end;
+
+		  label incomecat='Income Categories based on 2016 HUD Limit for Family of 4';
+
+run;
+
+proc sort data= hhtype_1_&year.;
+by serial;
+run;
+
+proc sort data= nonrelatehh_&year.;
+by serial;
+run;
+
+data hhtype_&year.;
+merge hhtype_1_&year. nonrelatehh_&year. ;
+by serial;
+run;
+
+data hhtype_&year.;
+set hhtype_&year.;
+if hhtype in (4,5,6,7) then do;  /*non family*/
```

Comment:
The coding in this section is wrong. Everyone ends up with HHcat=5. You need some **else if** statements where you are using just **if**'s to fix it.

Reply:
How about this?

‘’’

if hhtype in (4,5,6,7) then do;  /*non family*/
if numprec=1 then HHcat=1 ; /*single*/
else if nonrelatehh=1 then HHcat=4; /* non relate households*/ 
end;

else if hhtype in (1,2,3) then do; /*family household*/
    if hhtype=1 & numprec=2 then HHcat=2 ; /*couple without kid*/
else HHcat=3; /*other family*/
end;


if hhtype in (0, 9) then HHcat=.;

else HHcat=5;
‘’’

Reply:
The else HHcat=5 overwrites all the other categories assigned. I had to modify the code a bit to :
if hhtype in (4,5,6,7) then do; /_non family_/
if numprec=1 then HHcat=1 ; /_single_/
else if nonrelatehh=1 then HHcat=4; /* non relate households*/
end;

else if hhtype in (1,2,3) then do; /_family household_/
if hhtype=1 & numprec=2 then HHcat=2 ; /_couple without kid_/
else HHcat=3; /_other family_/
end;

else if hhtype in (0, 9) then do;
 HHcat=.;
end;
else HHcat=5;

But now no 5 is assigned....

Reply:
I think you want 
else if hhtype in (0, 9) then do;
HHcat=5;
end;

Where 5 is NA/undetermined. The final else statement doesn't do anything because all the values for hhtype are already accounted for. 
hhtype
hhtype

Reply:
@rpitingolo @maggiesu0725 As a follow up on the discussion of household types, part of the problem was relying upon the IPUMS HHTYPE var. Certain types of households got a HHTYPE=9 because IPUMS couldn't determine how they fit into their classification but they were still households that we want to count. Also, I found a few mistakes in the HHTYPE coding. In the end, I changed from using HHTYPE to creating our own variables to classify the households. This also allowed more consistent coding with 2000, where we did not have HHTYPE.